### PR TITLE
fix shebang in scripts installed with --symlink 

### LINKED ISSUE #285: flit install --symlink doest not honor --python for scripts it installs
Hi,

I have been experimenting with flit recently and I very much like it's approach for development installs (`--symlink`).

I noticed what could be a small issues with such installations.

I have flit installed globally (with pipx).

When I try to install my project with `pip install --symlink --python $(which python)` in a virtualenv, I notice the installed script uses flit's python as shebang instead of the python of the virtualenv.

I believe the problem arises in this `install_script`, called via `install_directly`:
https://github.com/takluyver/flit/blob/f5996dcee05656ca7a1e6ed2e5c82a6e911f4f10/flit/install.py#L171

### PR DESCRIPTION
**Pull request by sbidoul**
Fixes #285 

I also fixed a deprecation warning for an invalid escape sequence.

### CODE REVIEW

File: tests/test_install.py [hunk 1]
```diff
@@ -107,6 +111,8 @@ def test_symlink_other_python(self):
         assert_islink(self.tmpdir / 'site-packages2' / 'package1',
                       to=str(samples_dir / 'package1'))
         assert_isfile(self.tmpdir / 'scripts2' / 'pkg_script')
+        with (self.tmpdir / 'scripts2' / 'pkg_script').open() as f:
+            assert f.readline().strip() == "#!mock_python"
```
Comment (takluyver) [inline tests/test_install.py]: This makes me think we should probably do an `abspath` on it. I don't think we want shebangs with a relative path.

Comment (sbidoul) [reply tests/test_install.py]: That should probably go very early in the code, just after argument parsing.
That probably involves searching in `PATH` too, because `flit install --python python` just works today as way to install in the current active virtualenv.

Comment (sbidoul) [reply tests/test_install.py]: A reliable (although a bit inefficient) possibility could be something like this

```python
def _normalize_python(python):
    if python == sys.executable:
        return python
    return subprocess.check_output([python, "-c", "import sys; print(sys.executable)"]).strip()
```

Comment (sbidoul) [reply tests/test_install.py]: possibly with a shortcut if `isabs(python)`

Comment (sbidoul) [reply tests/test_install.py]: In any case, one could argue that flit works better with this PR than not, if only for people providing a full path in `--python`. So maybe it could be merged as is? Alternatively, if you think the above idea about normalizing `args.python` is valid, I'm happy to add a commit with it to this PR.

Comment (takluyver) [reply tests/test_install.py]: Yes, good points. In most cases I think `os.path.abspath(shutil.which(self.python))` should give the right answer, but it wouldn't surprise me if there's some odd corner cases with symlinks and things where that's different from running it in a subprocess and getting `sys.executable`.

I agree that this PR is a strict improvement, so I'm happy to merge this as if if you prefer.

**Comment by sbidoul** [comment]
Interestingly, one can see in the last commit that shutil.which and subprocess seem to produce different results.

image

**Comment by sbidoul** [comment]
Do I amended the last test. This should be complete now :)

**Comment by takluyver** [comment]
Thanks. That difference is strange - it looks like `shutil.which()` is getting the wrong path somehow.

**Comment by sbidoul** [comment]
Now if only pip uninstall would support those symlinks..

**Comment by sbidoul** [comment]
> Now if only pip uninstall would support those symlinks..

Updat:e: pip >= 19.3 now uninstalls symlinks. (https://github.com/pypa/pip/pull/6914)
